Feature/setup ci on travis

owodunni opened the pull request:
Very basic CI support, but we can build from here!

owodunni committed:
Add travis inorder to test functionality mentioned in README

owodunni committed:
Add master build status to README.md

owodunni commented:
Logon to https://travis-ci.org to see the lastest builds

hansfilipelo commented on .travis.yml:
```diff
@@ -0,0 +1,10 @@
+  - docker
+
+install:
+- cd src/bin
```
Install should be: 
```
make install
```
No cd necessary.

hansfilipelo commented on .travis.yml:
```diff
@@ -0,0 +1,10 @@
+- cd src/bin
+
+script:
+- ./dockrun wget https://dockrun.github.io
```
After install, remove:
```
./
```

hansfilipelo reviewed (commented):
See comments. Also we should possibly add steps for: 

- Removing container (dockrm wget)
- Try another package which can use the `-p` and `-r`command line arguments

owodunni commented:
Make install dosn’t work we don’t have access for writing on that part of
the file system..

On Wed, 5 Dec 2018 at 07:30, Hans-Filip Elo <notifications@github.com>
wrote:

> *@hansfilipelo* commented on this pull request.
>
> See comments. Also we should possibly add steps for:
>
>    - Removing container (dockrm wget)
>    - Try another package which can use the -p and -rcommand line arguments
>
> ------------------------------
>
> In .travis.yml
> <https://github.com/dockrun/dockrun/pull/1#discussion_r238943401>:
>
> > @@ -0,0 +1,10 @@
> +sudo: bash
> +
> +services:
> +  - docker
> +
> +install:
> +- cd src/bin
>
> Install should be:
>
> make install
>
> No cd necessary.
> ------------------------------
>
> In .travis.yml
> <https://github.com/dockrun/dockrun/pull/1#discussion_r238943491>:
>
> > @@ -0,0 +1,10 @@
> +sudo: bash
> +
> +services:
> +  - docker
> +
> +install:
> +- cd src/bin
> +
> +script:
> +- ./dockrun wget https://dockrun.github.io
>
> After install, remove:
>
> ./
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dockrun/dockrun/pull/1#pullrequestreview-181605909>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIK9UqqIWnq4Y6sRgqGfI_rdb94nJHe-ks5u12f0gaJpZM4ZCFV5>
> .
>

owodunni commented:
But I can look inte it and see if I’m doing something wrong.

On Wed, 5 Dec 2018 at 08:25, Alexander Poole <alex.o.poole@gmail.com> wrote:

> Make install dosn’t work we don’t have access for writing on that part of
> the file system..
>
> On Wed, 5 Dec 2018 at 07:30, Hans-Filip Elo <notifications@github.com>
> wrote:
>
>> *@hansfilipelo* commented on this pull request.
>>
>> See comments. Also we should possibly add steps for:
>>
>>    - Removing container (dockrm wget)
>>    - Try another package which can use the -p and -rcommand line
>>    arguments
>>
>> ------------------------------
>>
>> In .travis.yml
>> <https://github.com/dockrun/dockrun/pull/1#discussion_r238943401>:
>>
>> > @@ -0,0 +1,10 @@
>> +sudo: bash
>> +
>> +services:
>> +  - docker
>> +
>> +install:
>> +- cd src/bin
>>
>> Install should be:
>>
>> make install
>>
>> No cd necessary.
>> ------------------------------
>>
>> In .travis.yml
>> <https://github.com/dockrun/dockrun/pull/1#discussion_r238943491>:
>>
>> > @@ -0,0 +1,10 @@
>> +sudo: bash
>> +
>> +services:
>> +  - docker
>> +
>> +install:
>> +- cd src/bin
>> +
>> +script:
>> +- ./dockrun wget https://dockrun.github.io
>>
>> After install, remove:
>>
>> ./
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/dockrun/dockrun/pull/1#pullrequestreview-181605909>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AIK9UqqIWnq4Y6sRgqGfI_rdb94nJHe-ks5u12f0gaJpZM4ZCFV5>
>> .
>>
>

hansfilipelo commented:
> Make install dosn’t work we don’t have access for writing on that part of the file system..


It is possible to use another prefix for installation than the default one (`/usr/local`). 

```
PREFIX=/path/to/install make install
```

This should probably be added to the documentation, but I have not yet added it since this is also fairly standard procedure on POSIX compliant systems.

owodunni commented:
Sounds like just what we need!

On Wed, 5 Dec 2018 at 08:33, Hans-Filip Elo <notifications@github.com>
wrote:

> Make install dosn’t work we don’t have access for writing on that part of
> the file system..
>
> It is possible to use another prefix for installation than the default one
> (/usr/local).
>
> PREFIX=/path/to/install make install
>
>
> This should probably be added to the documentation, but I have not yet
> added it since this is also fairly standard procedure on POSIX compliant
> systems.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dockrun/dockrun/pull/1#issuecomment-444387129>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AIK9UkQMuh3TBO722OSCJ0nQkcwRoZc_ks5u13a1gaJpZM4ZCFV5>
> .
>

Commit:
Run make install with sudo

owodunni committed:
Set language to bash

This gives us a minimal container more can be read
here https://docs.travis-ci.com/user/languages/minimal-and-generic/

owodunni commented:
Turns out i had my head up my ass and that sudo is a thing...

owodunni commented:
Go ahead and merge when you feel like it!

hansfilipelo commented:
Great. I do see that we possibly need to test uninstall as well (`sudo make purge`). 

How does Travis execution work? Will it fail the test if one of the commands fail (this is what we want)?

owodunni commented:
Everything that is run under the script tag in the .travis.yml needs to pass for the build to pass. 

If we want to do tests such as uninstall and verify that the correct folder is removed the best way is to do a test bash script.

If we want to keep it simple we can just run make purge and see that it dosn't chrash.

hansfilipelo commented:
> If we want to do tests such as uninstall and verify that the correct folder is removed the best way is to do a test bash script.

This is what we want.

edit: Files to check for are: 
```
$PREFIX/bin/dockrun
$PREFIX/bin/dockrun
$PREFIX/lib/dockrun-common
```

owodunni commented:
Is there something more that should be tested while we are at it?

hansfilipelo commented:
This will do for a start. I do like to add some more tests for each command line argument (`-p`, `-h` and `-r`) - but I think testing install/uninstall is a good start.

owodunni committed:
Add test for uninstall

owodunni committed:
Run scripts under different stages

This makes it easy for us to see what works and what failes

owodunni committed:
Add name to builds

owodunni commented:
We now test that everything is uninstalled.

Build status can be viewed here:
https://travis-ci.org/dockrun/dockrun

hansfilipelo committed:
Tests purge by checking images left after purging

- Also follows project variable naming conventions
